### PR TITLE
Laravel components v8

### LINKED ISSUE #2538: Upgrade Illuminate components to 8.x
Let's upgrade Laravel to 8.x. Just sitting on 6.x will eventually cause Flarum to lag behind, possibly even outside of the LTS scope which is harmful to the product especially now that we close in on stable.

Laravel isn't really moving that fast anymore in terms of changes (yes they do tag more versions, but the actual changes are rather limited compared to diffs between 4 - 5 - 6), so the higher we can bump our dependency the more likely it is that we can keep up and reduce maintenance overhead on extension developers when we do.

Will look into this as soon as I can and once #2507 is merged.

### PR DESCRIPTION
**Pull request by luceos**
**Fixes #2538 **

Still needs to be tested inside the skeleton.

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.
- [x] Backend changes: tests are green (run `composer test`).

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

**Companion PRs**

- [ ] https://github.com/flarum/tags/pull/121

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -77,7 +80,6 @@
         "psr/http-server-middleware": "^1.0",
         "s9e/text-formatter": "^2.3.6",
         "symfony/config": "^4.3.4",
-        "symfony/console": "^4.3.4",
```
Comment (askvortsov1) [inline composer.json:80]: Where is this coming from if it's not explicitly required? We still use the Command class

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
On the Laravel upgrade guide, a dependence on Symfony 5 is listed as one of the high impact changes, but we don't seem to have changed our symfony version at all: https://laravel.com/docs/7.x/upgrade#symfony-5-related-upgrades. Symfony 5 seems to be implicitly required for console though. Are there downsides to having separate versions of symfony for different symfony components?

It seems like the most impactful change would be the deprecation of the Translator interface that's used commonly in extensions.

File: composer.json (hunk 1 above)
Comment (luceos) [reply composer.json:80]:
```
illuminate/console   v8.25.0  requires  symfony/console (^5.1.4)
illuminate/database  v8.25.0  requires  symfony/console (^5.1.4)
```

**Comment by luceos** [comment]
> 
> 
> On the Laravel upgrade guide, a dependence on Symfony 5 is listed as one of the high impact changes, but we don't seem to have changed our symfony version at all: https://laravel.com/docs/7.x/upgrade#symfony-5-related-upgrades. Symfony 5 seems to be implicitly required for console though. Are there downsides to having separate versions of symfony for different symfony components?
> 
> It seems like the most impactful change would be the deprecation of the Translator interface that's used commonly in extensions.

I still need to go over testing inside the skeleton as mentioned in OP. I bet there are a ton of other issues to fix before this all works ;)

File: composer.json (hunk 1 above)
Comment (askvortsov1) [reply composer.json:80]: If we're using it explicitly, I'd prefer to have the dependency listed explicitly as well.

**Comment by askvortsov1** [comment]
@luceos I tested this with a skeleton install, and made the above 2 commits, which mitigate most issues with extensions that use the translator.

Aside from that, there are a few issues we won't be able to compensate for:

- Extensions requiring fof/console will break, as that depends on illuminate/console v6, which depends on symfony console v4, which conflicts with core's use of symfony console v5.
- The signature of `setKeysForSaveQuery` on line 65 of TagState in tags needs the typehint removed to comply with Laravel's interface / abstract class.

Other than that, I haven't found issues in the things I tried.

File: composer.json [hunk 2]
```diff
@@ -76,25 +79,26 @@
         "psr/http-server-handler": "^1.0",
         "psr/http-server-middleware": "^1.0",
         "s9e/text-formatter": "^2.3.6",
-        "symfony/config": "^4.3.4",
-        "symfony/console": "^4.3.4",
-        "symfony/event-dispatcher": "^4.3.4",
+        "symfony/config": "^v5.2.2",
+        "symfony/console": "^5.2.2",
```
Comment (askvortsov1) [inline composer.json:83]: This should probably also have a v for consistency?

File: composer.json [hunk 3]
```diff
@@ -29,8 +32,8 @@
             "email": "matthew@kilgore.dev"
         },
         {
-          "name": "Alexander (Sasha) Skvortsov",
-          "email": "askvortsov@flarum.org"
+            "name": "Alexander (Sasha) Skvortsov",
```
Comment (askvortsov1) [inline composer.json:35]: These changes should be reverted, since individual names were removed from this file recently.

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Found one very minor formatting issue in composer.json

File: composer.json (hunk 2 above)
Comment (luceos) [reply composer.json:61]: Removed all v's because they add no value and no other dep has it.

File: composer.json (hunk 3 above)
Comment (luceos) [reply composer.json]: rebased

**Review by askvortsov1 — APPROVED** [review]

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Looks like there's a few more phpunit related changes needed to the tests before this can be merged

**Review by askvortsov1 — APPROVED** [review]

**Review by tankerkiller125 — APPROVED** [review]
Now that the test are good I'm happy with this

**Pull request merged**